src/charm.py

```diff
@@ -954,6 +976,70 @@ def _core_reconciliation(self):
         if self._current_wp_config() is not None:
             self._start_server()
 
+    def _wp_theme_list(self):
+        """List all installed WordPress themes"""
+        process = self._run_wp_cli(["wp", "theme", "list", "--format=json"], timeout=600)
+        if process.return_code != 0:
+            raise RuntimeError("wp theme list command failed: " + process.stderr)
+        try:
+            return json.loads(process.stdout)
+        except json.decoder.JSONDecodeError:
+            raise RuntimeError(
+                "wp theme list command failed, stdout is not json: " + repr(process.stdout)
+            )
+
+    def _wp_theme_install(self, theme):
+        """Install WordPress theme"""
+        process = self._run_wp_cli(
+            # --force will overwrite any installed version of the theme,
+            # without prompting for confirmation
+            ["wp", "theme", "install", theme, "--force"],
+            timeout=600,
+            combine_stderr=True
+        )
+        if process.return_code != 0:
+            raise RuntimeError("wp theme list command failed: " + process.stdout)
+
+    def _wp_theme_delete(self, theme):
+        """Uninstall WordPress theme"""
+        process = self._run_wp_cli(
+            ["wp", "theme", "delete", theme, "--force"],
```

Comment:
Should this always use `--force`? Are there any risks with using force to uninstall a theme?

Reply:
I think it will cause some trouble when uninstall a currently activated theme without "--force", but the theme activation is not managed by juju, so force remove the theme here.